src/test/java/co/rsk/federate/bitcoin/BtcTransactionBuilder.java

```diff
@@ -0,0 +1,105 @@
+package co.rsk.federate.bitcoin;
+
+import co.rsk.bitcoinj.core.Address;
+import co.rsk.bitcoinj.core.BtcTransaction;
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.core.TransactionInput;
+import co.rsk.bitcoinj.core.TransactionOutPoint;
+import co.rsk.bitcoinj.core.TransactionOutput;
+import co.rsk.bitcoinj.core.TransactionWitness;
+import co.rsk.bitcoinj.script.Script;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class BtcTransactionBuilder {
+
+    private final NetworkParameters networkParameters;
+    private final Map<Integer, TransactionWitness> transactionWitnesses = new HashMap<>();
+    private final List<TransactionInput> inputs = new ArrayList<>();
+    private final List<TransactionOutput> outputs = new ArrayList<>();
+
+    public BtcTransactionBuilder(NetworkParameters networkParameters) {
+        this.networkParameters = networkParameters;
+    }
+
+    public BtcTransactionBuilder withInput(TransactionInput transactionInput) {
+        inputs.add(transactionInput);
+        return this;
+    }
+
+    public BtcTransactionBuilder withInput(TransactionOutput transactionOutput) {
+        TransactionInput transactionInput = new TransactionInput(networkParameters, null,
+            new byte[]{}, transactionOutput.getOutPointFor(), transactionOutput.getValue());
+        inputs.add(transactionInput);
+        return this;
+    }
+
+    public BtcTransactionBuilder withWitness(int inputIndex,
+        TransactionWitness transactionWitness) {
+        transactionWitnesses.put(inputIndex, transactionWitness);
+        return this;
+    }
+
+    public BtcTransactionBuilder withOutput(Coin amount, Address address) {
+        outputs.add(
+            new TransactionOutput(networkParameters, null, amount, address)
+        );
+        return this;
+    }
+
+    public BtcTransaction build() {
+        BtcTransaction btcTransaction = new BtcTransaction(networkParameters);
+        addInputs(btcTransaction);
+        addOutputs(btcTransaction);
+        return btcTransaction;
+    }
+
+    private void addInputs(BtcTransaction btcTransaction) {
+        inputs.forEach(transactionInput -> {
+            int idx = btcTransaction.getInputs().size();
+            btcTransaction.addInput(transactionInput);
+            if (transactionWitnesses.containsKey(idx)) {
+                btcTransaction.setWitness(idx, transactionWitnesses.get(idx));
+            }
+        });
+    }
+
+    private void addOutputs(BtcTransaction btcTransaction) {
+        outputs.forEach(btcTransaction::addOutput);
+    }
+
+    public class InputBuilder {
```

Comment:
Making `InputBuilder` a static nested class might be a good idea here. Since it does not require an instance of `BtcTransactionBuilder` to be instantiated and does not need to access any non-static members of `BtcTransactionBuilder`.

Reply:
If I make it static, I will need to pass the `networkParameters` in the creation of an `InputBuilder` object. Also, it opens the possibility of creating an `InputBuilder` with a different `networkParameters` than the one used in the `TransactionBuilder`.

Reply:
but creating an input builder with a different network than the transaction builder wont be right i think

Reply:
I see, then another idea could be to create a method in `BtcTransactionBuilder` that ensures that the `InputBuilder` is created with the same network, maintaining consistency. Something like this, wdyt? 

```java
    public InputBuilder createInputBuilder(Coin amount) {
        return new InputBuilder(networkParameters, amount);
    }
```

Reply:
@apancorb  good solution. I like it.

Reply:
Done